lms/static/js/edxnotes/views/toggle_notes_factory.js

```diff
@@ -0,0 +1,73 @@
+;(function (define, undefined) {
+'use strict';
+define([
+    'jquery', 'underscore', 'backbone', 'gettext', 'js/edxnotes/views/visibility_decorator'
+], function($, _, Backbone, gettext, EdxnotesVisibilityDecorator) {
+    var ToggleNotesView = Backbone.View.extend({
+        events: {
+            'click .action-toggle-notes': 'toogleHandler'
+        },
+
+        errorMessage: gettext("We couldn't <show/hide> your notes. Make sure that you are connected to the Internet, and then try refreshing the page."),
```

Comment:
@srpearce @polesye I'm not sure about the error message now, it may be confusing. It will appear when the ```visibility``` state is not saved properly to the server. But the checkbox will toggle and the notes on the page will effectively show or hide. Only on reload will things be unchanged. So perhaps something like:

```We couldn't save the visibility state of your notes to our servers. Make sure that you are connected to the Internet, and then try refreshing the page.```